### PR TITLE
Use #create_credential_login instead of the deprecated #report_auth_info at auxiliary/capture/server/mssql

### PR DESCRIPTION
**Pull request by red0xff**
My attempt at fixing #16474 .

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use auxiliary/server/capture/mssql`
- [ ] ...
- [ ] **Verify**  The warning is gone.
- [ ] **Verify** The module still works as intended.

### CODE REVIEW

File: modules/auxiliary/server/capture/mssql.rb [hunk 1]
```diff
@@ -264,19 +264,18 @@ def mssql_get_hash(arg = {})
...
-        :host  => arg[:ip],
-        :port => 445,
-        :sname => 'smb_client',
-        :user => user,
-        :pass => domain + ":" +
+      report_cred(
+        ip: ip,
+        port: 445,
+        user: user,
+        sname: 'smb_client',
```
Comment (adfoster-r7) [inline modules/auxiliary/server/capture/mssql.rb:271]: Would this be `mssql_client` or similar 👀

**Comment by adfoster-r7** [comment]
Thanks for looking into this! 🎉 

I ran into some issues using john with the ntlmv2 hash

### basic creds

I was able to confirm basic creds are working:
```
msf6 auxiliary(server/capture/mssql) > 
[*] MSSQL LOGIN 192.168.123.128:44722  / <empty>
[*] MSSQL LOGIN 192.168.123.128:44724 user / pass
creds
Credentials
===========

host             origin           service                  public  private  realm  private_type  JtR Format
----             ------           -------                  ------  -------  -----  ------------  ----------
192.168.123.128  192.168.123.128  1433/tcp (mssql_client)          <empty>         Password      
192.168.123.128  192.168.123.128  1433/tcp (mssql_client)  user    pass            Password      
```

### windows authentication

When trying windows auth from mssql management studio via `choco install sql-server-management-studio`

image

msfconsole capturing:

```
msf6 auxiliary(server/capture/mssql) > 
[*] 2022-04-27 12:46:23 +0100
NTLMv2 Response Captured from 192.168.123.151 
DOMAIN: DESKTOP-N3MAG5R USER: admin 
LMHASH:Disabled LM_CLIENT_CHALLENGE:Disabled
NTHASH:b3ff9f11d0e741603a63c70d49388232 NT_CLIENT_CHALLENGE:0101000000000000ed19b96b2c5ad8010f96fb8be7611d5c000000000200120041004e004f004e0059004d004f00550053000100120041004e004f004e0059004d004f00550053000400120061006e006f006e0079006d006f00750073000300120061006e006f006e0079006d006f00750073000800300030000000000000000100000000200000030131faed2f21b0cd9fd754c1c8685caa8165c3b81b9db416533da197706be30a00100000000000000000000000000000000000090034004d005300530051004c005300760063002f004200460053002d004d00420050002d0039003900300035003a0031003400330033000000000000000000


msf6 auxiliary(server/capture/mssql) > creds
Credentials
===========

host             origin           service               public  private                                                                                              realm  private_type        JtR Format
----             ------           -------               ------  -------                                                                                              -----  ------------        ----------
192.168.123.151  192.168.123.151  445/tcp (smb_client)  admin   DESKTOP-N3MAG5R:000000000000000000000000000000000000000000000000:b3ff9f11d0e741603a63c7 (TRUNCATED)         Nonreplayable hash  netntlmv2

msf6 auxiliary(server/capture/mssql) > 
```

### Verifying  `john` with stored creds

The john format isnt' output to console by default which is a small improvement we could make to the logging, but I was able to grab the john hash output with `creds -v`

But it looks like we're not generating the right format for john:

```
➜  ~ john ./hash.txt --wordlist=./words.txt  --format=netntlmv2
Using default input encoding: UTF-8
No password hashes loaded (see FAQ)
```

A quick glance suggests the username is missing from the hash, and it looks like the challenge is in the wrong place too? 🤔 
```
DESKTOP-N3MAG5R:000000000000000000000000000000000000000000000000:b3ff9f11d0e741603a63c70d493882320101000000000000ed19b96b2c5ad8010f96fb8be7611d5c000000000200120041004e004f004e0059004d004f00550053000100120041004e004f004e0059004d004f00550053000400120061006e006f006e0079006d006f00750073000300120061006e006f006e0079006d006f00750073000800300030000000000000000100000000200000030131faed2f21b0cd9fd754c1c8685caa8165c3b81b9db416533da197706be30a00100000000000000000000000000000000000090034004d005300530051004c005300760063002f004200460053002d004d00420050002d0039003900300035003a0031003400330033000000000000000000:1122334455667788
```

Comparing to the output of Responder:
```
[+] Listening for events...                                                                              

[MSSQL] NTLMv2 Client   : ::ffff:192.168.123.151
[MSSQL] NTLMv2 Username : DESKTOP-N3MAG5R\admin
[MSSQL] NTLMv2 Hash     : admin::DESKTOP-N3MAG5R:6091a63daae32b9d:6CD72873F8FCCF841A9A9581EBD948E1:0101000000000000BBDF7286305AD8015F86B20BEEDD05490000000002000800560053004800350001001E00570049004E002D004E00580030005900590032004F0036004C00490058000400140056005300480035002E004C004F00430041004C0003003400570049004E002D004E00580030005900590032004F0036004C00490058002E0056005300480035002E004C004F00430041004C000500140056005300480035002E004C004F00430041004C000800300030000000000000000100000000200000030131FAED2F21B0CD9FD754C1C8685CAA8165C3B81B9DB416533DA197706BE30A0010000000000000000000000000000000000009003A004D005300530051004C005300760063002F003100390032002E003100360038002E003100320033002E003100320038003A0031003400330033000000000000000000 
```

I verified responder's output works with john as expected:
```
➜  ~ john ./hash.txt --wordlist=./words.txt  --format=netntlmv2
Using default input encoding: UTF-8
Loaded 1 password hash (netntlmv2, NTLMv2 C/R [MD4 HMAC-MD5 32/64])
Will run 4 OpenMP threads
Press 'q' or Ctrl-C to abort, almost any other key for status
password123      (admin)     
1g 0:00:00:00 DONE (2022-04-27 08:16) 100.0g/s 600.0p/s 600.0c/s 600.0C/s password..testing
Use the "--show --format=netntlmv2" options to display all of the cracked passwords reliably
Session completed. 
```

**Comment by red0xff** [comment]
I wonder which format the original module was using (I probably didn't introduce that bug, as I didn't change anything in the hash identification code).

I've updated the hash formats, everything works fine with john the ripper for me, I tested both NTLMv2 and NTLMv1, but I'm not sure if other edge cases exist (also, didn't test `NTLM_2_SESSION_RESPONSE`, but if the comment is right, it should work the same as `NTLMv1`.

**Comment by adfoster-r7** [comment]
## Release notes

Fixes a deprecation warning in the `auxiliary/capture/server/mssql` warning as a outputting a valid johntheripper format for offline password cracking
